feat(pkg/gomodulepath): add tests for paths with versions

Fenixoff opened the pull request:
Now, if you try to run `starport app github.com/alice/voter --sdk-version="launchpad"`, you will get project named "alice", not "voter".

fadeev commented:
@Fenixoff thanks for reporting this! The issue has been fixed in v0.13.1:

https://github.com/tendermint/starport/releases/tag/v0.13.1

Fenixoff commented:
Sorry, I noticed this after I added the request, so I marked it as a draft. Perhaps the tests I added may be helpful. Or just close this request, please.

ilgooz reviewed (approved):
This is perfect, thank you @Fenixoff! :ok_hand:
